src/users/components/organisation-access-permissions/organisation-access-permissions.component.ts

```diff
@@ -0,0 +1,284 @@
+      "organisationProfileId": "SOLICITOR_PROFILE",
+      "accessTypeId": "34",
+      "enabled": true
+    }
+  ]
+  `;
+
+  // todo: remove above when we have the real data and remove the JSON Parse below when real data is ready
+  @Output() public selectedPermissionsChanged = new EventEmitter<JurisdictionPermissionViewModel[]>();
+
+  public jurisdictions: TempJurisdicationModel[] = JSON.parse(this.jurisdictionsExample) as TempJurisdicationModel[];
+  public userAccessTypes: UserAccessType[] = JSON.parse(this.userAccessTypesExample) as UserAccessType[];
+
+  public permissions: JurisdictionPermissionViewModel[];
+  public jurisdictionPermissionsForm: FormGroup<AccessForm>;
+
+  private onDestory$ = new Subject<void>();
+
+  constructor(private fb: FormBuilder, private cdRef: ChangeDetectorRef,) {
+  }
+
+  ngOnInit(): void {
+    this.permissions = this.createPermissionsViewModel();
+    this.selectedPermissionsChanged.emit(this.permissions);
+    this.createFormAndPopulate();
+    this.subscribeToAccessTypesChanges();
+  }
+
+  ngOnDestroy(): void {
+    this.onDestory$.next();
+    this.onDestory$.complete();
+  }
+
+  get jurisdictionsFormArray(): FormArray<FormGroup<JurisdictionPermissionViewModelForm>> {
+    return this.jurisdictionPermissionsForm.controls.jurisdictions;
+  }
+
+  public createPermissionsViewModel() : JurisdictionPermissionViewModel[] {
+    return this.jurisdictions.map((jurisdiction) => {
+      const accessTypes = jurisdiction.accessTypes
+        .sort((a, b) => a.displayOrder - b.displayOrder)
+        .filter((accessType) => accessType.display)
+        .map((accessType) => {
+          const accessTypePermissionViewModel:AccessTypePermissionViewModel = {
+            accessTypeId: accessType.accessTypeId,
+            enabled: accessType.accessDefault,
+            display: accessType.display,
+            description: accessType.description,
+            accessMandatory: accessType.accessMandatory
+          };
+          const userAccessType = this.userAccessTypes.find((ua) => ua.accessTypeId === accessType.accessTypeId && ua.jurisdictionId === jurisdiction.jurisdictionid);
+          if (userAccessType) {
+            accessTypePermissionViewModel.enabled = userAccessType.enabled;
+          }
+          return accessTypePermissionViewModel;
+        });
+
+      const permission:JurisdictionPermissionViewModel = {
+        jurisdictionId: jurisdiction.jurisdictionid,
+        jurisdictionName: jurisdiction.jurisdictionName,
+        accessTypes: accessTypes
+      };
+      return permission;
+    });
+  }
+
+  private subscribeToAccessTypesChanges() {
+    this.jurisdictionsFormArray.controls.forEach((jurisdictionGroup: FormGroup<JurisdictionPermissionViewModelForm>) => {
+      this.createPermissionChangeObservableForGroup(jurisdictionGroup).pipe(takeUntil(this.onDestory$)).subscribe((permissions) => {
+        this.selectedPermissionsChanged.emit(permissions);
+        console.log(permissions);
```

Comment:
Can console log be removed?

Reply:
if you don't mind I was hoping to hold onto this until all the moving parts are finished
it'll be removed once the other components for the manage user page are done

Reply:
All good